front/app/community/[siren]/components/FicheSubventions/Trends.tsx

```diff
@@ -0,0 +1,223 @@
+'use client';
+
+import { PureComponent, useState } from 'react';
+
+import { Subvention } from '@/app/models/subvention';
+import { Switch } from '@/components/ui/switch';
+import { formatCompactPrice, formatNumber } from '@/utils/utils';
+import {
+  Bar,
+  BarChart,
+  LabelList,
+  Legend,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+
+import DownloadButton from './DownloadButton';
+
+type FormattedDataTrends = {
+  annee: number;
+  montant: number;
+  nombre: number;
+  budget?: number;
+};
+
+const CustomTooltip = ({ active, payload, label }: any) => {
+  if (active && payload && payload.length) {
+    const montantEntry = payload.find((entry: any) => entry.name === 'montant');
+    const budgetEntry = payload.find((entry: any) => entry.name === 'stackedBudget');
+    const budget = montantEntry.value + budgetEntry.value;
+    const tauxPublication =
+      montantEntry && budgetEntry
+        ? ((montantEntry.value / (montantEntry.value + budgetEntry.value)) * 100).toFixed(1)
+        : '0';
+
+    return (
+      <div className='rounded-lg border border-gray-200 bg-white p-4 shadow-lg'>
+        <p className='font-semibold text-gray-900'>{label}</p>
+        <div className='mt-2 space-y-1'>
+          {tauxPublication !== 'NaN' && (
+            <div className='flex items-center gap-2 pt-1'>
+              <span className='text-sm text-gray-600'>Taux de publication</span>
+              <span className='text-sm font-medium text-gray-900'>{tauxPublication}%</span>
+            </div>
+          )}
+          {payload.map((entry: any, index: number) => (
+            <div key={index} className='flex items-center gap-2'>
+              <div className='h-3 w-3 rounded-full' style={{ backgroundColor: entry.color }} />
+              <span className='text-sm text-gray-600'>
+                {entry.name === 'montant' && 'Subventions publiées'}
+                {entry.name === 'stackedBudget' && 'Budget des subventions'}
+              </span>
+              <span className='text-sm font-medium text-gray-900'>
+                {entry.name === 'montant' && formatCompactPrice(entry.value)}
+                {entry.name === 'stackedBudget' && formatCompactPrice(budget)}
+              </span>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  }
+  return null;
+};
+
+const renderLabel = (props: any) => {
+  const { x, y, width, value } = props;
+  return (
+    <text
+      x={x + width / 2}
+      y={y - 10}
+      fill='#4e4e4e'
+      textAnchor='middle'
+      dominantBaseline='middle'
+      fontSize='16'
+    >
+      {value ? formatCompactPrice(value) : ''}
+    </text>
+  );
+};
+
+export default function Trends({ data }: { data: Subvention[] }) {
+  const [subventionsCountDisplayed, setSubventionsCountDisplayed] = useState(false);
+
+  function trends(data: Subvention[]) {
```

Comment:
```suggestion
  function mapSubventionsToTrends(data: Subvention[]) {
```

Ou `calculateTrends` peut etre

Reply:
Done